app/data_model/app_models.py

```diff
@@ -0,0 +1,95 @@
+import datetime
+
+from dateutil.tz import tzutc
+from marshmallow import Schema, fields, post_load, pre_dump
+
+
+class QuestionnaireState:
+    def __init__(self, user_id, state_data, version, created_at=None, updated_at=None):
+        self.user_id = user_id
+        self.state_data = state_data
+        self.version = version
+        self.created_at = created_at or datetime.datetime.now(tz=tzutc())
```

Comment:
Won't this set the `created_at` timestamp both on the way in and out of the database? This shouldn't be an issue given that all entities will have been saved with these values - but could hide a bug in future?

Reply:
IRL: Agreed it would be neater to get the database layer to do this, however that involves using dynamodbs `update_item` rather than `put_item` .  This gets a bit cumbersome as you have to construct an expression of fields that need updating and a way of specifying fields that should be excludes.  I did start down this path but changed my mind when we hit an error due to one of our fields, `data` on `SubmittedResponse` being a reserved keyword (fine for put apparently, but not to put in an update expression).  I didn't want to get sucked into doing a field migration at the same time as adding the new tables.